src/ethereum/trace.py

```diff
@@ -0,0 +1,86 @@
+"""
+.. _trace:
+
+EVM Trace
+^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Defines the functions required for creating evm traces during execution.
+"""
+
+from typing import Any
+
+
+def capture_tx_start(env: Any) -> None:
```

Comment:
I'd recommend using `object` instead of `Any` for the arguments to these.

Reply:
Using `object` created issues because mypy keeps complaining about the existence of certain attributes in `object` type.

Reply:
That is expected, because `object` is the most generic type. You'll want to do some runtime checking on the object in `ethereum_spec_tools` to get at the properties.

Something like:

```python
from typing import runtime_checkable, Protocol

@runtime_checkable
class Foo(Protocol):
    property_a: int
    property_b: int

def bar(arg: object) -> int:
    assert isinstance(arg, Foo)
    return arg.property_a
```

Note that using structural typing at runtime (with `Protocol` and `isinstance`) is much slower than normal Python inheritance, so use it judiciously.

Reply:
Updated